core/capabilities/targets/write_target.go

```diff
@@ -117,68 +96,47 @@ func (cap *EvmWrite) Execute(ctx context.Context, request capabilities.Capabilit
 	if inputs.Report == nil {
 		// We received any empty report -- this means we should skip transmission.
 		cap.lggr.Debugw("Skipping empty report", "request", request)
-		callback := make(chan capabilities.CapabilityResponse)
-		go func() {
-			// TODO: cast tx.Error to Err (or Value to Value?)
-			callback <- capabilities.CapabilityResponse{
-				Value: nil,
-				Err:   nil,
-			}
-			close(callback)
-		}()
-		return callback, nil
+		return success(), nil
 	}
 	cap.lggr.Debugw("WriteTarget non-empty report - attempting to push to txmgr", "request", request, "reportLen", len(inputs.Report), "reportContextLen", len(inputs.Context), "nSignatures", len(inputs.Signatures))
 
 	// TODO: validate encoded report is prefixed with workflowID and executionID that match the request meta
 
-	// construct forwarder payload
-	calldata, err := forwardABI.Pack("report", common.HexToAddress(reqConfig.Address), inputs.Report, inputs.Context, inputs.Signatures)
-	if err != nil {
+	// Check whether value was already transmitted on chain
+	queryInputs := struct {
+		Receiver            string
```

Comment:
You might need to add ReportID here but let's merge this PR first and then handle new meta fields separately.